tools/src/wyvern/tools/parsing/coreparser/ASTBuilder.java

```diff
@@ -13,7 +13,7 @@
 	/** if type is null, it will be inferred*/
 	public AST valDecl(String name, Type type, AST exp, FileLocation loc);
 	public AST varDecl(String name, Type type, AST exp, FileLocation loc);
-	public AST defDecl(String name, Type type, List args, AST body, boolean isClassDef, FileLocation loc);
+	public AST defDecl(String name, Type type, List<Token> generics, List args, AST body, boolean isClassDef, FileLocation loc);
```

Comment:
This should be a List<String>, not a List<Token>.  Tokens only belong in the lexer.